baybe/recommenders/pure/bayesian/base.py

```diff
@@ -35,39 +27,27 @@ class BayesianRecommender(PureRecommender, ABC):
     surrogate_model: Surrogate = field(factory=GaussianProcessSurrogate)
     """The used surrogate model."""
 
-    acquisition_function_cls: Literal[
-        "PM", "PI", "EI", "UCB", "qPI", "qEI", "qUCB", "VarUCB", "qVarUCB"
-    ] = field(default="qEI")
+    acqf: AcquisitionFunction = field(
+        converter=convert_acqf, factory=qExpectedImprovement, kw_only=True
+    )
     """The used acquisition function class."""
 
-    _acquisition_function: Optional[AcquisitionFunction] = field(
-        default=None, init=False
-    )
+    _botorch_acqf = field(default=None, init=False)
     """The current acquisition function."""
 
-    def _get_acquisition_function_cls(
-        self,
-    ) -> Callable:
-        """Get the actual acquisition function class.
+    acquisition_function_cls: bool = field(default=None)
+    "Deprecated! Raises an error when used."
 
-        Returns:
-            The debotorchized acquisition function class.
-        """
-        mapping = {
-            "PM": PosteriorMean,
-            "PI": ProbabilityOfImprovement,
-            "EI": ExpectedImprovement,
-            "UCB": partial(UpperConfidenceBound, beta=1.0),
-            "qEI": qExpectedImprovement,
-            "qPI": qProbabilityOfImprovement,
-            "qUCB": partial(qUpperConfidenceBound, beta=1.0),
-            "VarUCB": partial(UpperConfidenceBound, beta=100.0),
-            "qVarUCB": partial(qUpperConfidenceBound, beta=100.0),
-        }
-        fun = debotorchize(mapping[self.acquisition_function_cls])
-        return fun
-
-    def setup_acquisition_function(
+    @acquisition_function_cls.validator
+    def _validate_deprecated_argument(self, _, value) -> None:
+        """Raise a DeprecationError if the tolerance flag is used."""
+        if value is not None:
+            raise DeprecationError(
+                "Passing 'acquisition_function_cls' to the constructor is deprecated. "
+                "The flag has been renamed to 'acqf'."
+            )
+
+    def setup_acqf(
```

Comment:
I'd actually prefer the full name instead of the abbreviation but would also be fine with this one if the two of you prefer it.

Reply:
No discussion needed since we simply go with what will be decided in the other thread, but keeping it open as a reminder to change the name if it becomes necessary

Reply:
resolving, please see other thread